go/vt/vttablet/onlineddl/executor.go

```diff
@@ -983,9 +984,38 @@ func (e *Executor) validateAndEditAlterTableStatement(ctx context.Context, alter
 		return true, nil
 	}
 	if err := sqlparser.Walk(validateWalk, alterTable); err != nil {
-		return err
+		return alters, err
+	}
+	alters = append(alters, alterTable)
+	// Handle ADD FULLTEXT KEY statements
+	countAddFullTextStatements := 0
+	for i := range alterTable.AlterOptions {
+		switch opt := alterTable.AlterOptions[i].(type) {
+		case *sqlparser.AddIndexDefinition:
+			if opt.IndexDefinition.Info.Fulltext {
+				if countAddFullTextStatements > 0 {
+					// We've already got one ADD FULLTEXT KEY. We can't have another
+					// in the same statement
+					alterTable.AlterOptions[i] = nil
+
+					extraAlterTable := &sqlparser.AlterTable{
+						Table:        alterTable.Table,
+						AlterOptions: []sqlparser.AlterOption{opt},
+					}
+					alters = append(alters, extraAlterTable)
+				}
+				countAddFullTextStatements++
+			}
+		}
 	}
-	return nil
+	for i := 0; i < len(alterTable.AlterOptions); {
+		if alterTable.AlterOptions[i] == nil {
+			alterTable.AlterOptions = append(alterTable.AlterOptions[0:i], alterTable.AlterOptions[i+1:]...)
+		} else {
+			i++
+		}
+	}
```

Comment:
Guess there's not really a much more simple way to get these out of the slice? Alternatively I guess the previous loop could build a new one and we replace that at the end, but dunno if that looks much clearer?

Reply:
Honestly I spent an hour over this silly thing; there is no canonical way in `golang`, that I can find, for removing elemtns from a slice while iterating it. I don't know what's the least evil way of doing it.

Reply:
@shlomi-noach Yeah, have solved it in the past with allocating a same sized slice and then copy into that instead vs. trying to remove from an existing slice. But this also works.